### PR TITLE
[Windows] Switch to 0x0A00 (Windows 10) as a primary version and fix build with new target.

### PR DESCRIPTION
**Pull request by bruvzg**
- Fixes build with target version set to 0x0A00 (Windows 10).
- Adds version check depending on "d3d12" flag.
- Sets default target to 0x0A00 (Windows 10).

See https://github.com/godotengine/godot/issues/88466

### CODE REVIEW

File: platform/windows/detect.py [hunk 1]
```diff
@@ -182,13 +182,14 @@ def get_opts():
...
-        # Targeted Windows version: 7 (and later), minimum supported version
+        # Targeted Windows version: 10 (and later), minimum supported version
         # XP support dropped after EOL due to missing API for IPv6 and other issues
         # Vista support dropped after EOL due to GH-10243
+        # 7, 8 and 8.1 support dropped after EOL due to Direct3D 12 rendering driver addition
         (
             "target_win_version",
-            "Targeted Windows version, >= 0x0601 (Windows 7)",
-            "0x0601",
+            "Targeted Windows version, >= 0x0A00 (Windows 10)",
```
Comment (akien-mga) [inline platform/windows/detect.py:191]: The `>=` part is meant to show the min supported version, which doesn't seem to be Windows 10. IIUC we still support down to Windows 7 if passing `0x601`.

Comment (bruvzg) [reply platform/windows/detect.py]: Updated description to mention both.

**Comment by bruvzg** [comment]
Actually, I have looked at DX loading, and doing it dynamically might be much easier than I was thinking, probably less than 10 functions, so this might be unnecessary.

**Comment by bruvzg** [comment]
Superseded by https://github.com/godotengine/godot/pull/88496, we can keep Windows 7/8.1 support for now.

**Pull request closed without merging**